rt_gene_model_training/dataset_manager.py

```diff
@@ -0,0 +1,164 @@
+                    subject['y'].append(img_lbl)
+                except:
+                    print('Failure loading pair!')
+
+            if self.random_subset:
+                np.random.seed(42)
+                num_samples = int(len(subject['y']) * self.random_subset)
+                indices = np.random.choice(len(subject['y']), num_samples, False)
+                subject['x'] = [np.array(right_inputs)[indices], np.array(left_inputs)[indices]]
+                subject['y'] = np.array(subject['y'])[indices].tolist()
+            else:
+                subject['x'] = [np.array(right_inputs), np.array(left_inputs)]
+        return subject
+
+    def load(self):
+        self.folds = {}
+        self.subjects = {}
+
+        with open(self.csv_subjects) as csvfile:
+            csv_rows = csv.reader(csvfile)
+            for row in tqdm(csv_rows):
+                subject_id = int(row[0])
+                csv_labels = row[1]
+                left_folder = row[2]
+                right_folder = row[3]
+                fold_type = row[4]
+                fold_id = int(row[5])
+
+                if fold_type == 'training':
+                    print('\nsubject ' + str(subject_id) + ' is loading...')
+                    csv_filename = self.csv_subjects.split('/')[-1]
+                    csv_labels = self.csv_subjects.replace(csv_filename, csv_labels)
+                    left_folder = self.csv_subjects.replace(csv_filename, left_folder)
+                    right_folder = self.csv_subjects.replace(csv_filename, right_folder)
+                    self.subjects[subject_id] = self.load_one_subject(csv_labels, left_folder, right_folder)
+                    if fold_id not in self.folds.keys():
+                        self.folds[fold_id] = []
+                    self.folds[fold_id].append(subject_id)
+                    
+                elif fold_type == 'discarded':
+                    print('\nsubject ' + str(subject_id) + ' is discarded.')
+                    
+                elif fold_type == 'validation':
+                    print('\nsubject ' + str(subject_id) + ' is ignored (validation fold).')
+                    
+        print(self.folds)
+        print(self.subjects.keys())
+
+    def get_folds(self, fold_ids):
+        fold = {}
+        subject_list = list(itertools.chain(*[self.folds[fold_id] for fold_id in fold_ids]))
+        all_x_left = [self.subjects[subject_id]['x'][0] for subject_id in subject_list]
+        all_x_right = [self.subjects[subject_id]['x'][1] for subject_id in subject_list]
+        all_y = [np.array(self.subjects[subject_id]['y']) for subject_id in subject_list]
+        fold['x'] = [np.concatenate(all_x_left), np.concatenate(all_x_right)]
+        fold['y'] = np.concatenate(all_y).tolist()
+        
+        fold['positive'] = 1 #TODO
+        fold['negative'] = 1 #TODO
+        return fold
+
+
+
+    
+if __name__ == '__main__':
+    csv_subjects = '/home/icub/Kevin/RT-GENE_dataset_eyes/subjects.csv'
```

Comment:
Do not use absolute paths, instead use relative paths (using script_path or something, see RT-GENE code) to ensure it works on different machines.

Reply:
It seems like this is some debugging code here anyway, can we remove the `__main__` in this file?